pkg/auth/handler/wire.go

```diff
@@ -0,0 +1,38 @@
+//+build wireinject
+
+package handler
+
+import (
+	"net/http"
+
+	"github.com/google/wire"
+
+	"github.com/skygeario/skygear-server/pkg/auth"
+	"github.com/skygeario/skygear-server/pkg/auth/dependency/authn"
+	"github.com/skygeario/skygear-server/pkg/core/db"
+	"github.com/skygeario/skygear-server/pkg/core/handler"
+	"github.com/skygeario/skygear-server/pkg/core/validation"
+)
+
+func provideLoginHandler(
+	requireAuthz handler.RequireAuthz,
+	v *validation.Validator,
+	ap LoginAuthnProvider,
+	tx db.TxContext,
+) http.Handler {
```

Comment:
I guess the return type of this function and `auth.MakeHandler` are problematic. Wire injects by type but in `/pkg/auth/handler` we have more than one `http.Handler`.

Reply:
For each handler inject, there'll two function in `wire.go`:
- `provideXXX`: instantiate handler using injected dependencies, returning `http.Handler`
- `newXXX`: specifies available dependencies and instantiate handler, returning `http.Handler`

Since each `newXXX` will specify only their corresponding `provideXXX` in the dependency set, there is only one provider for `http.Handler` (i.e. `provideXXX`), so there would be no conflict.